src/v/schema/tests/fake_registry.cc

```diff
@@ -51,19 +51,30 @@ schema::fake_store::get_schema_definition(ppsr::schema_id id) {
     }
     throw std::runtime_error("unknown schema id");
 }
+
+void schema::fake_registry::maybe_throw_injected_failure() const {
+    if (_inject_failures) {
+        throw std::runtime_error("Injected failure in fake_registry");
+    }
+}
+
 ss::future<ppsr::canonical_schema_definition>
 schema::fake_registry::get_schema_definition(ppsr::schema_id id) const {
+    maybe_throw_injected_failure();
```

Comment:
maybe this answers my other question about exception handling?

Reply:
Responded in the other comment. But yea it'd be nice if this were an outcome type